numpy/lib/index_tricks.py

```diff
@@ -589,6 +590,336 @@ def __iter__(self):
     next = __next__
 
 
+def fproduct(*iterables, **kwargs):
+    """
+    Like itertools.product, but in fortran order
+    """
+    return (i[::-1] for i in itertools.product(*iterables[::-1], **kwargs))
+
+
+class ndrange(_nx.collections_abc.Sequence):
+    """
+    An N-dimensional range object that returns an iterator (or reversed
+    iterator) tthat can produce a sequence of tuples from
+    start (inclusive tuple) to stop (exclusive tuple) by step.
+
+    start defaults to a tuple of zeros and stop is omited!
+
+    When step is given, it specifies the increment (or decrement).
+
+    ndrange is designed to behave much like the Python3 ``range`` object.
+    See `python class range documentation
+    <https://docs.python.org/3/library/stdtypes.html#range>`_ for more details.
+    About valid parameter types and behaviour peculiarities. One notable
+    difference is that the the iterator returned by ``ndrange`` will always
+    return tuples as opposed to singletons.
+
+    Unlike ``ndindex``, ``ndrange`` is not an iterator.
+    You should prefer this ``ndrange`` object over ``ndindex`` as ``ndrange``
+    allows for multi-dimensional slicing.
```

Comment:
I think this comment belongs in the nditer docstring, but isn't needed in the ndrange one.

Reply:
ok. adapted.